flink-ml-lib/src/test/java/org/apache/flink/ml/common/datastream/AllReduceUtilsTest.java

```diff
@@ -0,0 +1,145 @@
+        }
+
+        @Test
+        public void testAllReduce() throws Exception {
+            StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+            env.setParallelism(parallelism);
+            DataStream<double[]> elements =
+                    env.fromParallelCollection(
+                                    new NumberSequenceIterator(1L, parallelism),
+                                    BasicTypeInfo.LONG_TYPE_INFO)
+                            .map(
+                                    x -> {
+                                        double[] res = new double[numElements];
+                                        for (int i = 0; i < res.length; i++) {
+                                            res[i] = i;
+                                        }
+                                        return res;
+                                    });
+
+            AllReduceUtils.allReduce(elements)
+                    .addSink(
+                            new SinkFunction<double[]>() {
+                                @Override
+                                public void invoke(double[] value, Context context) {
+                                    assertEquals(value.length, numElements);
+                                    for (int i = 0; i < value.length; i++) {
+                                        assertEquals(value[i], i * parallelism, 1e-9);
+                                    }
+                                }
+                            });
+
+            env.execute();
+        }
+    }
+
+    /** Non-parameterized test for {@link org.apache.flink.ml.common.datastream.AllReduceUtils}. */
+    public static class NonParameterizedTest {
+
+        @Test
+        public void testAllReduceWithMoreThanOneArray() {
+            try {
+                StreamExecutionEnvironment env =
+                        StreamExecutionEnvironment.getExecutionEnvironment();
+                env.setParallelism(parallelism);
+                DataStream<double[]> elements =
+                        env.fromParallelCollection(
+                                        new NumberSequenceIterator(1L, parallelism),
+                                        BasicTypeInfo.LONG_TYPE_INFO)
+                                .flatMap(
+                                        new FlatMapFunction<Long, double[]>() {
+                                            @Override
+                                            public void flatMap(
+                                                    Long value, Collector<double[]> out) {
+                                                out.collect(new double[100]);
+                                                out.collect(new double[100]);
+                                            }
+                                        });
+
+                AllReduceUtils.allReduce(elements).addSink(new SinkFunction<double[]>() {});
+                env.execute();
```

Comment:
Should we call `Assert.fail(...)` here? Otherwise, the test will pass even if this method does not throw exception.

Same for other tests.